### PR TITLE
Initial commit of System.IO partial facade

### PR DESCRIPTION
**Pull request by FiveTimesTheFun**

### CODE REVIEW

File: src/System.IO/System.IO.sln [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+Global
+	GlobalSection(SolutionConfigurationPlatforms) = preSolution
+		Debug|Any CPU = Debug|Any CPU
+		Release|Any CPU = Release|Any CPU
+	EndGlobalSection
+	GlobalSection(ProjectConfigurationPlatforms) = postSolution
+		{07390899-C8F6-4E83-A3A9-6867B8CB46A0}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
+		{07390899-C8F6-4E83-A3A9-6867B8CB46A0}.Debug|Any CPU.Build.0 = Debug|Any CPU
+		{07390899-C8F6-4E83-A3A9-6867B8CB46A0}.Release|Any CPU.ActiveCfg = Release|Any CPU
+		{07390899-C8F6-4E83-A3A9-6867B8CB46A0}.Release|Any CPU.Build.0 = Release|Any CPU
```
Comment (stephentoub) [inline src/System.IO/System.IO.sln]: What's "ActiveCfg" and "Build" in this context?

**Comment by stephentoub** [comment]
Is there a test project to go along with this?  I realize at present there's no much to test, just the single exception type.

Otherwise, LGTM.

File: src/System.IO/System.IO.sln (hunk 1 above)
Comment (FiveTimesTheFun) [inline src/System.IO/System.IO.sln]: They map the solution's configuration to the project's configuration (in the UI, Solution -> Properties -> Configuration Properties -> Configuration)

This means that for a debug solution configuration, we build the debug project.  For a release solution configuration, we build the release project.

**Comment by FiveTimesTheFun** [comment]
Thanks Stephen :)  Yes tests are coming - we want to add tests for the types that are forwarded to mscorlib in addition to the exception.

**Comment by FiveTimesTheFun** [comment]
Build machine appears to be hung...

@dotnet-bot test this please.

**Comment by FiveTimesTheFun** [comment]
Build machine appears to be hung...

@dotnet-bot retest this please

File: src/System.IO/System.IO.sln (hunk 1 above)
Comment (stephentoub) [inline src/System.IO/System.IO.sln]: Ah, sorry, I was looking at the wrong section.

**Comment by FiveTimesTheFun** [comment]
@dotnet-bot test this please.

File: src/System.IO/src/project.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
+  "dependencies": {
+    "Microsoft.DotNet.CoreCLR": "1.0.0-prerelease"
+  }
```
Comment (weshaggard) [inline src/System.IO/src/project.json]: @mellinoe @FiveTimesTheFun does having this reference mean we automatically reference mscorlib? I was kind of still expecting to see the mscorlib reference somewhere in the project file to make it clear.

**Pull request merged**